src/views/open-api-docs-view/components/open-api-sidebar-contents/index.tsx

```diff
@@ -3,42 +3,87 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
-import { OperationGroup, OperationProps } from 'views/open-api-docs-view/types'
+import { useMemo } from 'react'
+import { useRouter } from 'next/router'
+// Lib
+import { useActiveSection } from 'lib/hash-links/use-active-section'
+// Components
+import { SidebarNavMenuItem } from 'components/sidebar/components'
+// Types
+import type { OpenApiNavItem } from 'views/open-api-docs-view/types'
+// Styles
+import s from './open-api-sidebar-contents.module.css'
 
 /**
  * Renders sidebar contents for OpenApiDocsView.
- *
- * TODO: implement this component.
- * Designs show a list of navigational items
- * that link to the corresponding #operation-slug on the page.
- * We also want to highlight the current navigational item.
- *
- * Note: we likely also want to re-use this component to render
- * the OpenApiDocsView level of the corresponding mobile nav.
  */
 export function OpenApiSidebarContents({
-	operationGroups,
+	navItems,
 }: {
-	operationGroups: OperationGroup[]
+	navItems: OpenApiNavItem[]
 }) {
+	const { asPath } = useRouter()
+
+	/**
+	 * Note next/router `asPath` returns *with* the `#hash`.
+	 *
+	 * Here, we remove the hash, so that we can compare to the `fullPath`
+	 * of each nav item, which are not expected to have hashes,
+	 * to the URL path that we're on.
+	 *
+	 * Note that domain does not matter, as we're just grabbing the hash
+	 */
+	const urlPathname = useMemo(
+		() => new URL(asPath, 'https://www.example.com').pathname,
+		[asPath]
+	)
+
+	/**
+	 * Build an array of section slugs,
+	 * and pass these to `useActiveSection`.
+	 */
+	const sectionSlugs = useMemo(() => {
+		return (
+			navItems
+				.map((item) => {
+					// Only grab slugs from link items
+					if (!('fullPath' in item)) {
+						return null
+					}
+					// Note that domain does not matter, we're just grabbing the hash
+					const { hash } = new URL(item.fullPath, 'https://www.example.com')
+					return hash.replace('#', '')
+				})
+				// Filter out any null or empty string values
+				.filter((slug) => typeof slug === 'string' && slug !== '')
+		)
+	}, [navItems])
+	const activeSection = useActiveSection(sectionSlugs)
+
+	/**
+	 * Highlight any active matched sidenav items.
+	 * These could match `#activeSection` or the full pathname.
+	 */
+	const navItemsWithActive = useMemo(() => {
```

Comment:
❓do we want to have multiple active nav items?

Reply:
I think so! Feels necessary in order to highlight both the active `/path`, and the active `#operation-hash`, without additional logic or bespoke-ness:

![CleanShot 2023-07-31 at 08 22 03@2x](https://github.com/hashicorp/dev-portal/assets/4624598/e71f2834-373d-49f5-a8a5-ff3dbfcc689b)